### ISSUE TITLE
Angular 14 doesn't build library with multiple entry points

### ISSUE THREAD

**Issue by mgameover** [angular/angular-cli#23293]

**Comment by alan-agius4** [comment]
The cause of the problem here is that there is an index file (https://github.com/mgameover/app-n-lib/blob/ng-14/projects/components/shared/index.ts ) which exports all the symbols but it is not mark as the entry-point.

This causes a recursive exports of itself, as now in version 14 when the entry-point is something other than an `index.ts` an index file is generated.

We should handle this case. As a workaround for the time being, you can move the contents of the index in the public-api file.

**Comment by alan-agius4** [comment]
It looks like the path mapping is pointing to the source of the library and not the built version in the dist.

**Comment by mgameover** [comment]
yes, path mapping points to source and that's done intentionally - changes in the lib are picked up automatically so you don't need to rebuild lib everyone.

**Comment by mgameover** [comment]
Hi @alan-agius4, 
As I can see from the [changelog](https://github.com/ng-packagr/ng-packagr/blob/main/CHANGELOG.md), the fix has been released in v14.0.1. 
However I don't see any changes. I tried to update the same repo, I can see that the version is changed to v14.0.1 but it doesn't solve the issue.

So my question is  - what is a proper way to handle this? 

- Remove index.ts
- or rename public-api.ts to index.ts and update ng-package.json
- or there will be another fix

**Comment by alan-agius4** [comment]
In your particular case the best action would be to rename public-api.ts to index.ts and update ng-package.json

**Comment by alan-agius4** [comment]
It actually looks like 14.0.1 didn't contain the change mentioned in the
ng-packagr. I will cut another ng-packagr release later this week.

**Comment by alan-agius4** [comment]
Ng-packagr version 14.0.2 has been released.

**Comment by dmitry-stepanenko** [comment]
For anyone coming across this with a slightly different error
```
Entry point 'my-lib' has an 'index.ts' parallel to the 'entryFilePath'. The 'entryFilePath' should be updated to point to the 'index.ts' file.
```

Make sure you don't have internal imports in the library, that point to the main `index.ts` file of the lib.
